wrappers/windres-wrapper.sh

```diff
@@ -1,3 +1,183 @@
+  coff                       COFF object
+Targets:
+  armv7-w64-mingw32
+  aarch64-w64-mingw32
+  i686-w64-mingw32
+  x86_86-w64-mingw
+EOF
+    exit 1
+}
+
+error() {
+    echo "${PROG}: error: $1" >&2
+    exit 1
+}
+
+quote() {
+    echo "$1" | sed -e "s|'|'\\\\''|g; 1s/^/'/; \$s/\$/'/"
+}
+
+INCLUDE=
+VERBOSE=${VERBOSE:-false}
+INPUT=-
+OUTPUT=/dev/stdout
+INPUT_FORMAT=rc
+OUTPUT_FORMAT=res
+TARGET=X64
+CODEPAGE=
+
+while [ $# != 0 ]; do
+    case "$1" in
+        "-i="*|"--input="*)
+            INPUT="${1#*=}";;
+        "-i"|"--input")
+            INPUT="${2}"; shift;;
+        "-o="*|"--output="*)
+            OUTPUT="${1#*=}";;
+        "-o"|"--output")
+            OUTPUT="${2}"; shift;;
+        "-J="*|"--input-format="*)
+            INPUT_FORMAT="${1#*=}";;
+        "-J"|"--input-format")
+            INPUT_FORMAT="${2}"; shift;;
+        "-O="*|"--output-format="*)
+            OUTPUT_FORMAT="${1#*=}";;
+        "-O"|"--output-format")
+            OUTPUT_FORMAT="${2}"; shift;;
+        "-F="*|"--target="*)
+            TARGET="${1#*=}";;
+        "-O"|"--output-format")
+            TARGET="${2}"; shift;;
+        "-I"|"--include-dir")
+            INCLUDE="${INCLUDE} ${1#*=}"
+            ;;
+        "-I="*|"--include-dir="*)
+            INCLUDE="${INCLUDE} ${2}"; shift
+            ;;
+        "-c"|"--codepage")
+            CODEPAGE="${1#*=}"
+            ;;
+        "-c="*|"--codepage="*)
+            CODEPAGE="${2}"; shift
+            ;;
+        "--preprocessor")
+            error "ENOSYS"
+            ;;
+        "--preprocessor-arg")
+            error "ENOSYS"
+            ;;
+        "-v"|"--verbose")
+            VERBOSE=true
+            ;;
+        "-V"|"--version")
+            print_version;;
+        "--help"|"-h")
+            print_help;;
+        "-"*)
+            error "unrecognized option: \`$1'"
+            ;;
+        *)
+            error "rip: \`$1'"
+    esac;
+    shift
+done
+
+if $VERBOSE; then
+    set -x
+fi
+
+
+TMPDIR="$(mktemp -d /tmp/windres.XXXXXXXXX)" || error "couldn't create temp dir"
+
+case "${INPUT_FORMAT}" in
+    "rc")
+        $CC -E -P -xc -DRC_INVOKED=1 "${INPUT}" -o "${TMPDIR}/post.rc" || error "preprocessor failed"
+        sed '/^#/d;/^$/d' "${TMPDIR}/post.rc" > "${TMPDIR}/in.rc"
+        case "${OUTPUT_FORMAT}" in
+            "res")
+                llvm-rc "${TMPDIR}/in.rc" /FO "${TMPDIR}/out.res"
+                cat "${TMPDIR}/out.res" > "${OUTPUT}"
+                ;;
+            "coff")
+                llvm-rc "${TMPDIR}/in.rc" /FO "${TMPDIR}/in.res"
+                llvm-cvtres "${TMPDIR}/in.res" /OUT:"${TMPDIR}/out.o"
+                cat "${TMPDIR}/out.o" > "${OUTPUT}"
+                ;;
+            *)
+                error "invalid output format: \`${OUTPUT_FORMAT}'"
+        esac
+        ;;
+    "res")
+        cat "${INPUT}" > "${TMPDIR}/in.rc"
+        /usr/local/llvm-mingw/bin/llvm-rc "${TMPDIR}/in.res" /OUT:"${TMPDIR}/out.o"
```

Comment:
Leftover absolute path.

Also, what does this conversion even do, I don't see llvm-rc accepting `/out:` parameters? I guess this case would be a res->coff conversion and this is supposed to be llvm-cvtres instead?

Reply:
Ah, my bad. This requires the above target parsing (and mapping to LLVM target names)

Reply:
This was meant for the /machine: comment